orttraining/orttraining/core/graph/pipeline_transformer.cc

```diff
@@ -189,7 +189,7 @@ Node* AddBackwardRecord(Graph& graph,
   // TODO: use full list instead of the first element after changining
   // topological sort to depth-first from inputs.
   std::vector<NodeArg*> sub_backward_leaf_node_args{backward_leaf_node_args[0]};
-  ConvertTensorToBoolSignal(graph, backward_leaf_node_args, input_args);
+  ConvertTensorToBoolSignal(graph, sub_backward_leaf_node_args, input_args);
```

Comment:
>sub_backward_leaf_node_args [](start = 35, length = 27)

It seems the one in FW may still take some memory, but I guess we can figure out later.